### ISSUE TITLE
[BUG] @repeat_every blocks app startup on version 0.6.0

### ISSUE THREAD

**Issue by jonathanhar** [dmontagu/fastapi-utils#305]
**Describe the bug**
On version 0.2.1 there is no problem, but when using 0.6.0 the app doesn't start. Attaching simple code to reproduce

**To Reproduce**
```python
import logging
from contextlib import asynccontextmanager

import uvicorn
from fastapi import FastAPI
from fastapi_utils.tasks import repeat_every

logging.basicConfig(level=logging.INFO, force=True)
logger = logging.getLogger(__name__)


@asynccontextmanager
async def lifespan(app: FastAPI):
    await do_something()
    yield


app = FastAPI(lifespan=lifespan)


@repeat_every(seconds=1, logger=logger)  # 30 Minutes
async def do_something() -> None:
    logger.info("Removing idle devcontainers...")
    return


@app.get("/")
async def health():
    return 'Success'


if __name__ == "__main__":
    uvicorn.run(app, host="localhost", port=8080)
```

**Expected behavior**
App should start

**Environment:**
 - OS: Tested on Linux and Windows
 - FastAPI Utils version 0.6.0
 - Python version 3.11

**Comment by dwdSF** [comment]
++, the app does not start. It also seems that the option `wait_first` does not work

**Comment by Vanderhoof** [comment]
Mommy always told me: lock your dependencies. But I never listened to her. Now I lost several hours on a deadline week figuring out why the server is not starting.
This is very unfortunate, but a life lesson nevertheless

**Comment by IgnacioHeredia** [comment]
Same issue here.

```console
    from fastapi_utils.tasks import repeat_every
  File "/home/iheredia/anaconda3/lib/python3.8/site-packages/fastapi_utils/__init__.py", line 4, in <module>
    from .cbv_base import Api, Resource, set_responses, take_init_parameters
  File "/home/iheredia/anaconda3/lib/python3.8/site-packages/fastapi_utils/cbv_base.py", line 5, in <module>
    from .cbv import INCLUDE_INIT_PARAMS_KEY, RETURN_TYPES_FUNC_KEY, _cbv
  File "/home/iheredia/anaconda3/lib/python3.8/site-packages/fastapi_utils/cbv.py", line 21, in <module>
    from typing_inspect import is_classvar
ModuleNotFoundError: No module named 'typing_inspect'
```

**Comment by jonesbusy** [comment]
Same issue. My app doesn't start after update to 0.6.0

**Comment by ollz272** [comment]
Same here, needs fixing asap

**Comment by mgkid3310** [comment]
Same here, and during my struggle to fix the issue I found that the typing_inspect module's imported within fastapi-restful but it's not listed as a dependency so it has to be installed separately via pip.

**Comment by juanggcc** [comment]
This issue broke all my apps that relied on fastapi-utils for cron. Really need to start locking my dependencies.

**Comment by mgkid3310** [comment]
Seems like await (https://github.com/dmontagu/fastapi-utils/blob/master/fastapi_utils/tasks.py#L77) is the cause of the problem, it should be asyncio.ensure_future or Thread().start()

**Comment by Keiishu** [comment]
I just tested replacing (https://github.com/dmontagu/fastapi-utils/blob/master/fastapi_utils/tasks.py#L77) with `asyncio.ensure_future(loop())` and it works.

**Comment by ollz272** [comment]
@Keiishu can you raise a PR?

**Comment by Keiishu** [comment]
@ollz272 Sure! Here it is: #308

**Comment by yuval9313** [comment]
I've seen another issue aimed to solve another problem that caused this issue, I intend to approve any PR made about it but I want the [former issue](https://github.com/yuval9313/FastApi-RESTful/pull/142) not to come back

**Comment by mgkid3310** [comment]
@yuval9313 The former issue and PR aimed to remove `@app.on_event("startup")` decorator for the repeated functions, but the following comments show mixed results. Plus, the `@app.on_event("startup")` itself was deprecated from FastAPI, and now `@asynccontextmanager` and `async def lifespan(app: FastAPI):` is used.
(the `@repeat` decorator is my in-house code but works similar to `@repeat_every` before the 0.6.0 bug)

``` py
@asynccontextmanager
async def lifespan(app: FastAPI):
	log.info(f'Starting {cfg.api_name} v{cfg.api_version} server')
	await scan_results()

	yield

	log.info(f'Shutting down {cfg.api_name} v{cfg.api_version} server')

@utils_common.repeat(seconds=60, delay=30)
async def scan_results():
	report.scan_results()

app = FastAPI(lifespan=lifespan)
```

So unlike the issue title is suggesting I don't think such bug (repeat working only once) will happen, and the tests from pytest have confirmed this.

**Comment by RamiAwar** [comment]
@yuval9313 I'm on 0.7.0, still facing issues:

File "/home/app/.venv/lib/python3.12/site-packages/fastapi_utils/cbv.py", line 21, in <module>
from typing_inspect import is_classvar
ModuleNotFoundError: No module named 'typing_inspect'

Am I missing something?

**Comment by RamiAwar** [comment]
@yuval9313 I'm on 0.7.0 (latest) and still facing issues:

```
    from fastapi_utils.tasks import repeat_every
  File ".../.venv/lib/python3.12/site-packages/fastapi_utils/__init__.py", line 4, in <module>
    from .cbv_base import Api, Resource, set_responses, take_init_parameters
  File ".../.venv/lib/python3.12/site-packages/fastapi_utils/cbv_base.py", line 5, in <module>
    from .cbv import INCLUDE_INIT_PARAMS_KEY, RETURN_TYPES_FUNC_KEY, _cbv
  File ".../.venv/lib/python3.12/site-packages/fastapi_utils/cbv.py", line 21, in <module>
    from typing_inspect import is_classvar
ModuleNotFoundError: No module named 'typing_inspect'
```

**Comment by RamiAwar** [comment]
Wait why is the installation docs mentioning another package name?

<img width="762" alt="image" src="https://github.com/dmontagu/fastapi-utils/assets/8523191/cdbce900-9695-430b-a280-47a6adc6a33c">

This is different on Github vs docs page.

**Comment by RamiAwar** [comment]
Okay fixed when I installed with [all]. Nevermind. But docs need correction!

`poetry add "fastapi-utils[all]"`

**Comment by yuval9313** [comment]
Sorry about the docs, the CI should've deployed the latest version with the fix, I'll check it soon!

**Comment by Gerrit-K** [comment]
I can confirm what @RamiAwar [commented](https://github.com/dmontagu/fastapi-utils/issues/305#issuecomment-2165519256). It still is an issue on `0.7.0`. Though, I think it's not related to the original issue, but rather just to [this comment](https://github.com/dmontagu/fastapi-utils/issues/305#issuecomment-2112087864), for which there is a separate issue: #318.

**Comment by mgkid3310** [comment]
> I can confirm what @RamiAwar [commented](https://github.com/dmontagu/fastapi-utils/issues/305#issuecomment-2165519256). It still is an issue on `0.7.0`. Though, I think it's not related to the original issue, but rather just to [this comment](https://github.com/dmontagu/fastapi-utils/issues/305#issuecomment-2112087864), for which there is a separate issue: #318.

It sould be ok when installed via `fastapi-utils[all]`, as [all] includes all optional dependencies including typing_inspect. Adding typing_inspect to standard install might be a good solution though.
